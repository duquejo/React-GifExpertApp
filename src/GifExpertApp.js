import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import PropTypes from 'prop-types';

export const GifExpertApp = ( { defaultCategories = [] }) => {

  const [ categories, setCategories ] = useState( defaultCategories );

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr/>
      <ol>
        { 
          categories.map( category => <GifGrid key={ category } category={ category }/> )
        }
      </ol>
    </>
  );
};

/**
 * Required Component Rule
 */
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};