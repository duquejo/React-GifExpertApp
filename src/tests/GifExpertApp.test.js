import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('<GifExpertApp /> unit tests', () => {
  
  test('Should render <GifExpertApp /> successfully', () => {
    
    const wrapper = shallow( <GifExpertApp /> );

    expect( wrapper ).toMatchSnapshot();
  });

  test('Should show a categories list', () => {
    
    const categories = [ 'One punch', 'Dragon Ball' ];
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
  });
  
});