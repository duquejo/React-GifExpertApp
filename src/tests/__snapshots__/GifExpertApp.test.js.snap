// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> unit tests Should render <GifExpertApp /> successfully 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp /> unit tests Should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
